contrib/inventory/nsot.py

```diff
@@ -0,0 +1,342 @@
+         }
+       }
+     },
+     "rtr_and_fw": {
+       "hosts": [
+         "test1.example.com",
+         "test2.example.com"
+       ],
+       "vars": {}
+     }
+   }
+
+
+   $ NSOT_INVENTORY_CONFIG=$PWD/test.yaml ansible_nsot --host test1 | jq '.'
+   {
+      "make": "MX80",
+      "site_id": 1,
+      "id": 107
+   }
+
+'''
+
+from __future__ import print_function
+import sys
+import os
+import pkg_resources
+import argparse
+import json
+import yaml
+from textwrap import dedent
+from pynsot.client import get_api_client
+from pynsot.app import HttpServerError
+from click.exceptions import UsageError
+
+from six import string_types
+
+def warning(*objs):
+        print("WARNING: ", *objs, file=sys.stderr)
+
+
+class NSoTInventory(object):
+    '''NSoT Client object for gather inventory'''
+
+    def __init__(self):
+        self.config = dict()
+        config_env = os.environ.get('NSOT_INVENTORY_CONFIG')
+        if config_env:
+            try:
+                config_file = os.path.abspath(config_env)
+            except IOError:  # If file non-existent, use default config
+                self._config_default()
+            except Exception as e:
+                sys.exit('%s\n' % e)
+
+            with open(config_file) as f:
+                try:
+                    self.config.update(yaml.safe_load(f))
+                except TypeError:  # If empty file, use default config
+                    warning('Empty config file')
+                    self._config_default()
+                except Exception as e:
+                    sys.exit('%s\n' % e)
+        else:  # Use defaults if env var missing
+            self._config_default()
+        self.groups = self.config.keys()
+        self.client = get_api_client()
+        self._meta = {'hostvars': dict()}
+
+    def _config_default(self):
+        default_yaml = '''
+        ---
+        routers:
+          query: deviceType=ROUTER
+        switches:
+          query: deviceType=SWITCH
+        firewalls:
+          query: deviceType=FIREWALL
+        '''
+        self.config = yaml.safe_load(dedent(default_yaml))
+
+    def do_list(self):
+        '''Direct callback for when ``--list`` is provided
+
+        Relies on the configuration generated from init to run
+        _inventory_group()
+        '''
+        inventory = dict()
+        for group, contents in self.config.items():
+            group_response = self._inventory_group(group, contents)
+            inventory.update(group_response)
+        inventory.update({'_meta': self._meta})
+        return json.dumps(inventory)
+
+    def do_host(self, host):
+        return json.dumps(self._hostvars(host))
+
+    def _hostvars(self, host):
+        '''Return dictionary of all device attributes
+
+        Depending on number of devices in NSoT, could be rather slow since this
+        has to request every device resource to filter through
+        '''
+        device = [i for i in self.client.devices.get()['data']['devices']
```

Comment:
To make this compatible w/ NSoT v1.x please remove `['data']['devices']`. The `.get()` call on its own is good.